core/optaplanner-constraint-streams/src/main/java/org/optaplanner/constraint/streams/tri/FilteringTriJoiner.java

```diff
@@ -37,4 +39,20 @@ public FilteringTriJoiner(TriPredicate<A, B, C> filter) {
         return filter;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
FilteringTriJoiner is final, so we can just do instanceof
```suggestion
        if (o == null || o instanceof FilteringTriJoiner) {
```

null can be cast to everything but is an instance of nothing, right?